controller/destination/listener.go

```diff
@@ -9,25 +9,47 @@ import (
 	coreV1 "k8s.io/api/core/v1"
 )
 
+type podsByIpFn func(string) ([]*coreV1.Pod, error)
+
 type updateListener interface {
 	Update(add []common.TcpAddress, remove []common.TcpAddress)
-	Done() <-chan struct{}
+	ClientClose() <-chan struct{}
+	ServerClose() chan struct{}
```

Comment:
tioli, we could make this api a bit more restrictive with something like:
```go
ClientClose() <-chan struct{}
ServerClose() <-chan struct{}
CloseServer()
...
func (c *collectUpdateListener) CloseServer() {
  close(c.stopCh)
}
```

Reply:
Good thinking. I decided to call the method to close the channel `Stop()` though.